opm/simulators/wells/WellInterfaceFluidSystem.cpp

```diff
@@ -594,25 +581,14 @@ checkMaxWGRLimit(const WellEconProductionLimits& econ_production_limits,
     auto wgr = [](const std::vector<double>& rates,
                   const PhaseUsage& pu) {
 
-        const double water_rate = rates[pu.phase_pos[Water]];
-        const double gas_rate = rates[pu.phase_pos[Gas]];
-
-        // both rate should be in the same direction
-        assert(water_rate * gas_rate >= 0.);
-
-        double water_gas_ratio = 0.;
-
-        if (gas_rate != 0.) {
-            water_gas_ratio = water_rate / gas_rate;
-        } else {
-            if (water_rate != 0.) {
-                water_gas_ratio = 1.e100; // big value to mark it as violated
-            } else {
-                water_gas_ratio = 0.0;
-            }
-        }
-
-        return water_gas_ratio;
+        const double water_rate = -rates[pu.phase_pos[Water]];
+        const double gas_rate = -rates[pu.phase_pos[Gas]];
+        if (water_rate <= 0.)
+            return 0.;
+        else if (gas_rate <= 0.)
+            return 1.e100; // big value to mark it as violated
```

Comment:
Do we do anything regarding to these big values? Should we just throw here?

Reply:
I don't think we should throw. We should handle cases with 0 or negative gas rate by closing the well. The return of the large values here is just to make sure we close the wells due to economic  reason.